File: SkateboardAS/Core/Interfaces/IWarehouseService.cs
using Core.Models;

namespace Core.Interfaces;

public interface IWarehouseService
{
    Task<CommandResult> PickItemAsync(string trayId, CancellationToken ct = default);
    Task<CommandResult> InsertItemAsync(string trayId, string name, CancellationToken ct = default);
    Task<Inventory> GetInventoryAsync(CancellationToken ct = default);
}

File: SkateboardAS/core/src/main/java/dk/sdu/core/api/IWarehouseService.java
package dk.sdu.core.api;

import dk.sdu.core.model.CommandResult;
import dk.sdu.core.model.Inventory;

public interface IWarehouseService {
    CommandResult pickItem(String trayId);
    CommandResult insertItem(String trayId, String name);
    Inventory getWarehouseInventory();
}
